easybuild/tools/filetools.py

```diff
@@ -250,69 +249,13 @@ def det_common_path_prefix(paths):
         return None
 
 
-def download_file(filename, url, path):
-    """Download a file from the given URL, to the specified path."""
-
-    _log.debug("Trying to download %s from %s to %s", filename, url, path)
-
-    timeout = build_option('download_timeout')
-    if timeout is None:
-        # default to 10sec timeout if none was specified
-        # default system timeout (used is nothing is specified) may be infinite (?)
-        timeout = 10
-    _log.debug("Using timeout of %s seconds for initiating download" % timeout)
-
-    # make sure directory exists
-    basedir = os.path.dirname(path)
-    mkdir(basedir, parents=True)
-
-    # try downloading, three times max.
-    downloaded = False
-    max_attempts = 3
-    attempt_cnt = 0
-    while not downloaded and attempt_cnt < max_attempts:
-        try:
-            # urllib2 does the right thing for http proxy setups, urllib does not!
-            url_fd = urllib2.urlopen(url, timeout=timeout)
-            _log.debug('response code for given url %s: %s' % (url, url_fd.getcode()))
-            write_file(path, url_fd.read())
-            _log.info("Downloaded file %s from url %s to %s" % (filename, url, path))
-            downloaded = True
-            url_fd.close()
-        except urllib2.HTTPError as err:
-            if 400 <= err.code <= 499:
-                _log.warning("URL %s was not found (HTTP response code %s), not trying again" % (url, err.code))
-                break
-            else:
-                _log.warning("HTTPError occured while trying to download %s to %s: %s" % (url, path, err))
-                attempt_cnt += 1
-        except IOError as err:
-            _log.warning("IOError occurred while trying to download %s to %s: %s" % (url, path, err))
-            attempt_cnt += 1
-        except Exception, err:
-            _log.error("Unexpected error occurred when trying to download %s to %s: %s" % (url, path, err))
-
-        if not downloaded and attempt_cnt < max_attempts:
-            _log.info("Attempt %d of downloading %s to %s failed, trying again..." % (attempt_cnt, url, path))
-
-    if downloaded:
-        _log.info("Successful download of file %s from url %s to path %s" % (filename, url, path))
-        return path
-    else:
-        _log.warning("Download of %s to %s failed, done trying" % (url, path))
-        return None
-
-
-def find_easyconfigs(path, ignore_dirs=None):
+def find_easyconfigs(path, ignore_dirs=[]):
```

Comment:
no, change that back, there's a good reason we don't use lists as default value anywhere

if you would (ever) append to `ignore_dirs` in this function body, you're modifying the default for the entire Python session

using mutable types as default values is highly disencouraged